streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamPartitionAssignor.java

```diff
@@ -119,6 +122,24 @@ public void configure(Map<String, ?> configs) {
         streamThread = (StreamThread) o;
         streamThread.partitionAssignor(this);
 
+        String userEndPoint = (String) configs.get(StreamsConfig.USER_ENDPOINT_CONFIG);
+        if (userEndPoint != null && !userEndPoint.isEmpty()) {
+            final String[] hostPort = userEndPoint.split(":");
+            if (hostPort.length != 2) {
+                log.warn("Config '{}' isn't in the correct host:pair format. This instance will not be available for discovery"
```

Comment:
Maybe, i was and am undecided. At first i thought exception, but then I thought "Do we really want to stop the app from running due to this?"

I'm on the fence - happy to change it, happy to leave it.